synapse/federation/federation_server.py

```diff
@@ -1411,15 +1363,15 @@ def server_acl_evaluator_from_event(acl_event: EventBase) -> "ServerAclEvaluator
         logger.warning("Ignoring non-list deny ACL %s", deny)
         deny = []
     else:
-        deny = [glob_to_regex(s) for s in deny if isinstance(s, str)]
+        deny = [s for s in deny if isinstance(s, str)]
```

Comment:
A shame that the spec doesn't force rejection of ACLs with non-string allow/denylist entries.

Reply:
Yes, but this isn't changing the logic (of ignoring them).